### ISSUE TITLE
Properly extract files (FamilyTree especially)

### ISSUE THREAD

**Issue by philippbayer** [openSNP/snpr#490]
Right now the preparsing job uses the `zip` gem to unzip uploaded files, but uploaded files can come in a variety of flavours. FTDNA just updated their file format to be gz.

To do:

- [ ] In the preparsing job, test more which file format the user has uploaded (zip, tar.gz, gz, flat text, rar?)
- [ ] then use the right gem to unzip instead of going for default zip (probably need to add a few gems as it looks like there's no 'generic' ruby gem for all kinds of compressions)

**Comment by raivivek** [comment]
Parsing some files in the data dump was not a smooth experience. There are file with names such as `user7170_file5562_yearofbirth_unknown_sex_unknown.23andme.txt` but are zipped in reality. Looks like a good bug to fix.

**Comment by gedankenstuecke** [comment]
Yep, this is a fix we really(tm) should get to. @philippbayer has already started working on this in #492 I think. @raivivek any chance you wanna pick up that PR and give it more testing etc? 😄

**Comment by raivivek** [comment]
Sure! I'll be happy to help iron it out.

**Comment by gedankenstuecke** [comment]
That would be great, thanks!
